flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -234,7 +262,11 @@ protected void initChannel(SocketChannel socketChannel) {
                     }
                 };
         NioEventLoopGroup group =
-                new NioEventLoopGroup(1, new ExecutorThreadFactory("flink-rest-client-netty"));
+                new NioEventLoopGroup(
+                        1,
+                        new ExecutorThreadFactory("flink-rest-client-netty"),
+                        SelectorProvider.provider(),
```

Comment:
```suggestion
                        // mimicking the default for NioEventLoopGroup(int, Executor) here because there is no constructor that allows us to change the SelectStrategy without specifying the SelectorProvider
                        SelectorProvider.provider(),
```
Shall we add a comment here that describes where `SelectorProvider.provider()` is coming from?